Arrowgene.Ddon.GameServer/Chat/Command/Commands/FinishQuestCommand.cs

```diff
@@ -61,17 +61,27 @@ public override void Execute(string[] command, GameClient client, ChatMessage me
                 IsHelpReward = quest.RewardParams.IsHelpReward,
                 IsPartyBonus = quest.RewardParams.IsPartyBonus,
             };
-            client.Party.SendToAll(completeNtc);
 
-            client.Party.QuestState.UpdatePriorityQuestList(_server, client, client.Party);
-
-            if (quest.ResetPlayerAfterQuest)
+            if (quest.IsPersonal)
             {
-                foreach (var memberClient in client.Party.Clients)
+                client.QuestState.CompleteQuest(quest.QuestScheduleId);
+                client.Send(completeNtc);
+            }
+            else
+            {
+                client.Party.QuestState.CompleteQuestProgress(quest.QuestScheduleId);
+                client.Party.QuestState.UpdatePriorityQuestList(client);
+                client.Party.SendToAll(completeNtc);
+
+                if (quest.ResetPlayerAfterQuest)
                 {
-                    _server.CharacterManager.UpdateCharacterExtendedParamsNtc(memberClient, memberClient.Character);
+                    foreach (var memberClient in client.Party.Clients)
```

Comment:
Technically this is only needed so far for the very first main story quest, but I guess it doesn't hurt to do it for everyone just in case we use it again.